packages/apps/graph-client/src/utils/hooks/use-parsed-blocks.ts

```diff
@@ -0,0 +1,52 @@
+import { useCallback, useState } from 'react';
+
+interface IBlock {
+  height: number;
+  chainid: number;
+}
+interface IUseParseBlocksReturn {
+  allBlocks: Record<number, IBlock[]>;
+  addBlocks: (blocks: IBlock[]) => void;
+}
+
+export function useParsedBlocks(): IUseParseBlocksReturn {
+  const [allBlocks, setAllBlocks] = useState<Record<number, IBlock[]>>({});
+
+  const addBlocks = useCallback(
```

Comment:
useCallback can be typed with a generic. Then you don't need to type the arguments

```suggestion
  const addBlocks = useCallback<(blocks: IBlock[]) => void>(
```